packages/base/src/mainview/mainView.tsx

```diff
@@ -593,24 +593,75 @@ export class MainView extends React.Component<IProps, IStates> {
     this._updateLayersImpl(layerIds);
   }
 
+  /**
+   * Updates the layers in the OL map based on the layer IDs.
+   *
+   * @param layerIds - An array of layer IDs that should be present on the map.
+   */
   private async _updateLayersImpl(layerIds: string[]): Promise<void> {
-    const mapLayers: BaseLayer[] = [];
-    for (const layerId of layerIds) {
+    // get layers that are currently on the OL map
+    const previousLayerIds = this.getLayerIDs();
+
+    // Iterate over the new layer IDs
+    for (let i = 0; i < layerIds.length; i++) {
```

Comment:
```suggestion
    for (let targetLayerPosition = 0; targetLayerPosition < layerIds.length; targetLayerPosition++) {
```

Consider a name that will provide more context to the reader?